bundler/lib/bundler/checksum.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module Bundler
+  class Checksum
+    attr_reader :name, :version, :platform
+    attr_accessor :checksum
+
+    SHA256 = /\Asha256-([a-z0-9]{64}|[A-Za-z0-9+\/=]{44})\z/.freeze
+
+    def initialize(name, version, platform, checksum = nil)
```

Comment:
i think it makes sense for `checksum` here to be an array of `checksums`, so we have the flexibility to add more hashing algorithms down the road